pkg/reconciler/v1alpha1/autoscaling/kpa_scaler.go

```diff
@@ -107,6 +111,11 @@ func (rs *kpaScaler) Scale(kpa *kpa.PodAutoscaler, desiredScale int32) error {
 	// When ServingState=Reserve (see above) propagates to us, then actually scale
 	// things to zero.
 	if kpa.Spec.ServingState != v1alpha1.RevisionServingStateActive {
+		// Delay the scale to zero until the LTT of "Active=False" is some time in the past.
```

Comment:
We should also be subtracting the grace period from the effective [scale-to-zero-threshold](https://github.com/knative/serving/blob/3978c72683609cd3a628ee1bdb28eaac303a9fdd/config/config-autoscaler.yaml#L71) so that the actual time to zero is close to what the Operator has configured.  (Per conversation with @evankanderson and @vaikas-google).

So with a `gracePeriod` of 2 minutes and `scale-to-zero-threshold` of 5 minutes, we should transition the Revision to reserve state after 3 minutes of inactivity.   This will also require a minimum `scale-to-zero-threshold` value, or at least a comment.

Reply:
Done